projects/common/src/telemetry/track/track.directive.ts

```diff
@@ -0,0 +1,60 @@
+import { fromEvent, Subscription } from 'rxjs';
+import { Directive, ElementRef, Input, OnChanges, OnDestroy } from '@angular/core';
+import { TypedSimpleChanges } from '../../utilities/types/angular-change-object';
+import { TrackUserEventsType } from '../telemetry';
+import { UserTelemetryImplService } from '../user-telemetry-impl.service';
+
+@Directive({
+  selector: '[htTrack]'
+})
+export class TrackDirective implements OnChanges, OnDestroy {
+  @Input('htTrack')
+  public userEvents: TrackUserEventsType[] = [TrackUserEventsType.Click];
+
+  @Input('htTrackLabel')
+  public label?: string;
+
+  private readonly subscription: Subscription = new Subscription();
+  private trackedEventLabel: string = '';
+
+  public constructor(
+    private readonly host: ElementRef,
+    private readonly userTelemetryImplService: UserTelemetryImplService
+  ) {}
+
+  public ngOnChanges(changes: TypedSimpleChanges<this>): void {
+    if (changes.userEvents) {
+      this.setupListeners();
+    }
+
+    if (changes.label) {
+      this.trackedEventLabel = this.label ?? (this.host.nativeElement as HTMLElement)?.tagName;
+    }
+  }
+
+  public ngOnDestroy(): void {
+    this.clearListeners();
+  }
+
+  private setupListeners(): void {
+    this.clearListeners();
```

Comment:
I can't remember, does a subscription allow new subscriptions to be added after unsubscribe is called?

Reply:
Yeah, it would immediately unsubscribe from it. Fixing now.